File: samples/MiscellaneousSamples/app/src/com/pdftron/android/pdfnetsdksamples/OutputTextView.java
//---------------------------------------------------------------------------------------
// Copyright (c) 2001-2018 by PDFTron Systems Inc. All Rights Reserved.
// Consult legal.txt regarding legal and license information.
//---------------------------------------------------------------------------------------

package com.pdftron.android.pdfnetsdksamples;

import android.app.Activity;
import android.widget.ScrollView;
import android.widget.TextView;

public class OutputTextView implements OutputListener {
    
    private final TextView mOutputTextView;
    private final ScrollView mOutputScrollView;
    private final Activity mActivity;
    
    public OutputTextView(TextView textView, ScrollView scrollView, Activity activity) {
        this.mOutputTextView = textView;
        this.mOutputScrollView = scrollView;
        this.mActivity = activity;
    }
    
    @Override
    public void print(final String output) {
        
        mActivity.runOnUiThread(new Runnable() {
            
            @Override
            public void run() {
                mOutputTextView.append(output);
                
                mOutputScrollView.post(new Runnable() {
                    // This is necessary to scroll the ScrollView to the bottom.
                    @Override
                    public void run() {
                        mOutputScrollView.fullScroll(ScrollView.FOCUS_DOWN);
                    }
                });
            }
        });
    }

    @Override
    public void println(final String output) {
        
        mActivity.runOnUiThread(new Runnable() {
            
            @Override
            public void run() {
                mOutputTextView.append(output + "\n");
                
                mOutputScrollView.post(new Runnable() {
                    // This is necessary to scroll the ScrollView to the bottom.
                    @Override
                    public void run() {
                        mOutputScrollView.fullScroll(ScrollView.FOCUS_DOWN);
                    }
                });
                
            }
        });
    }

    @Override
    public void print() {
        print("");
    }

    @Override
    public void println() {
        println("");
    }

    @Override
    public void println(StackTraceElement[] stackTrace) {
        for (int i = 0; i < stackTrace.length; i++) {
            println(stackTrace[i].toString());
        }
    }
}

File: samples/MiscellaneousSamplesKt/app/src/com/pdftron/android/pdfnetsdksamples/OutputTextView.kt
//---------------------------------------------------------------------------------------
// Copyright (c) 2001-2018 by PDFTron Systems Inc. All Rights Reserved.
// Consult legal.txt regarding legal and license information.
//---------------------------------------------------------------------------------------

package com.pdftron.android.pdfnetsdksamples

import android.app.Activity
import android.widget.ScrollView
import android.widget.TextView

class OutputTextView(private val mOutputTextView: TextView, private val mOutputScrollView: ScrollView, private val mActivity: Activity) : OutputListener {

    override fun print(output: String?) {

        mActivity.runOnUiThread {
            mOutputTextView.append(output)

            mOutputScrollView.post( // This is necessary to scroll the ScrollView to the bottom.
            {
                mOutputScrollView.fullScroll(ScrollView.FOCUS_DOWN)
            })
        }
    }

    override fun println(output: String?) {

        mActivity.runOnUiThread {
            mOutputTextView.append(output + "\n")

            mOutputScrollView.post( // This is necessary to scroll the ScrollView to the bottom.
            {
                mOutputScrollView.fullScroll(ScrollView.FOCUS_DOWN)
            })
        }
    }

    override fun print() {
        print("")
    }

    override fun println() {
        println("")
    }

    override fun println(stackTrace: Array<StackTraceElement>) {
        for (i in stackTrace.indices) {
            println(stackTrace[i].toString())
        }
    }
}
